src/goto-symex/complexity_limiter.cpp

```diff
@@ -211,3 +212,29 @@ void complexity_limitert::run_transformations(
     for(auto transform_lambda : violation_transformations)
       transform_lambda.transform(complexity_violation, current_state);
 }
+
+/// Amount of nodes \p expr contains, with a bound on how far to search.
+/// Starts with an existing count.
+/// \return Size of \p expr added to count without searching significantly
+/// beyond the supplied limit.
+static std::size_t
+bounded_expr_size(const exprt &expr, std::size_t count, std::size_t limit)
+{
+  const auto &ops = expr.operands();
+  count += ops.size();
+  for(const auto &op : ops)
+  {
+    if(count >= limit)
+    {
+      return count;
+    }
+    count = bounded_expr_size(op, count, limit);
```

Comment:
If we aren't going to fix this, can we at least have a comment saying this is potentially risky?

Reply:
Comment now added to the public member function.